Bugsnag/Helpers/BSGRunContext.m

```diff
@@ -378,9 +378,19 @@ static void LoadLastRunContext(int fd) {
 
 /// Truncates or extends the file to the size of struct BSGRunContext,
 /// maps it into memory, and sets the `bsg_runContext` pointer.
-static void ResizeAndMapFile(int fd) {
+static void ResizeAndMapFile(const char *path, int fd) {
```

Comment:
This feels weird passing in the path of a file that's already been opened. What about a function `openWithProtectionDisabled`, which takes a path and returns an fd?